src/tests/user.test.ts

```diff
@@ -1,44 +1,166 @@
+        (userService.createUser as jest.Mock).mockReturnValue(() => { throw new Error(); });
+        const newUser = await _createUserController(sequelizeMock())(undefined as any, undefined as any, undefined as any);
+        expect(newUser).toBeInstanceOf(Error);
+    });
 
-        (userService.getUserByID as jest.Mock).mockReturnValue(() => Promise.resolve(userData));
+    test('Create user with poorly formed email should throw a validation error', async () => {
+        (userService.createUser as jest.Mock).mockReturnValue(() => { throw new ValidationError('Invalid email', []); });
+        const newUser = await _createUserController(sequelizeMock())(user.badEmail, user.password, user.displayName);
+        expect(newUser).toBeInstanceOf(ValidationError);
+    });
 
-        const response = await _getUserByIDController(sequelizeMock)('8c2d50d1-9b1f-483c-b267-cecb929ffb97');
+    test('Create user with short display name should throw a validation error', async () => {
+        (userService.createUser as jest.Mock).mockReturnValue(() => { throw new ValidationError('Invalid display name', []); });
+        const newUser = await _createUserController(sequelizeMock())(user.badEmail, user.password, '');
+        expect(newUser).toBeInstanceOf(ValidationError);
+    });
 
-        expect(response).toEqual(userData);
+    test('Create user with long display name should throw a validation error', async () => {
+        (userService.createUser as jest.Mock).mockReturnValue(() => { throw new ValidationError('Invalid display name', []); });
+        const newUser = await _createUserController(sequelizeMock())(user.badEmail, user.password, '>30 chars aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa');
+        expect(newUser).toBeInstanceOf(ValidationError);
     });
+});
 
-    test('If the user does not exist, it should return undefined', async () => {
-        (userService.getUserByID as jest.Mock).mockReturnValue(() => Promise.resolve(undefined));
+describe('Authenticate User', () => {
+    test('Authenticating existing user should return the user', async () => {
+        (userService.authenticate as jest.Mock).mockReturnValue(() => Promise.resolve(userResponse));
+        const authUser = await _authenticateController(sequelizeMock())(user.email, user.password);
+        expect(authUser).toEqual(userResponse);
+    });
 
-        const response = await _getUserByIDController(sequelizeMock)('8c2d50d1-9b1f-483c-b267-cecb929ffb97');
+    test('Authenticating non-existing user should return undefined', async () => {
+        (userService.authenticate as jest.Mock).mockReturnValue(() => Promise.resolve(undefined));
+        const authUser = await _authenticateController(sequelizeMock())(undefinedUser.email, undefinedUser.password);
+        expect(authUser).toBe(undefined);
+    });
 
-        expect(response).toBeUndefined();
+    test('Authenticating user with bad data should throw an error', async () => {
```

Comment:
This test description is not related to the test itself. This description would be good if we were testing `userService.authenticate` as it's that method's responsibility throw an error, not the controller's. The controller's responsibility to return the value of `userService.authenticate`, or an error it caught